### DISCUSSION TITLE
USB Audio interface

### DISCUSSION THREAD

**Discussion by saraverbeecke** [micropython/micropython#10842] [STM32 / Pyboard]
Does anyone know if it is feasible to send a multi-channel digital audio stream (like TDM format) to a PC using a STM32H7 using MicroPython? Therefor it should act as an USB audio interface, but I cannot find any MicroPython examples on this subject...

**Comment by garthk** [comment]
It's not yet one of the [examples](https://github.com/micropython/micropython/tree/master/examples/usb) but [check out this working USB audio input gist](https://gist.github.com/benagricola/ce97696df9377a9f9c38045b89769a5b)! :wave: @benagricola

**Comment by jonnor** [comment]
For the input side, I2S is pretty well-supported on STM32. See machine.I2S: https://docs.micropython.org/en/latest/library/machine.I2S.html

PDM and TDM are not supported at the moment, I believe.
